Integrations/integration-ThreatX.yml

```diff
@@ -0,0 +1,711 @@
+
+    @logger
+    def get_entity_risk(entity_id):
+        commands = {
+                'command': 'risk_changes',
+                'id': entity_id
+            }
+
+        return http_request('/entities', commands)
+
+    @logger
+    def get_entities_command():
+        entity_name = demisto.args().get('entity_name', None)
+        entity_id = demisto.args().get('entity_id', None)
+        entity_ip = demisto.args().get('entity_ip', None)
+        results = get_entities(entity_name, entity_id, entity_ip)
+        dbot_scores = []
+        output = []
+
+        for entity in results:
+            # Grab the entity risk so we can set the Dbot score for the Actor IPs
+            e_id = entity['id']
+            e_risk = get_entity_risk(e_id)[-1]['risk']
+            entity['risk'] = e_risk
+
+            iplist = []
+            for actor in entity['actors']:
+                # Convert decimal IP to dotted quad
+                ipint = actor['ip_address']
+                ipdot = '.'.join([str(ipint >> (i << 3) & 0xFF) for i in range(4)[::-1]])
+                iplist.append(ipdot)
+                actor['ip_address'] = ipdot
+                # Convert interval_time_start, interval_time_stop, fingerprint.last_seen to pretty time format
+                actor['interval_time_start'] = time.strftime("%Y-%m-%d %H:%M:%S", time.gmtime(actor['interval_time_start']))
+                actor['interval_time_stop'] = time.strftime("%Y-%m-%d %H:%M:%S", time.gmtime(actor['interval_time_stop']))
+                actor['fingerprint']['last_seen'] = time.strftime("%Y-%m-%d %H:%M:%S", time.gmtime(actor['fingerprint']['last_seen']))
+                # Set Dbot scores
+                if e_risk >= 70:
+                    dbot_scores.append({
+                            'Vendor' : 'ThreatX',
+                            'Indicator' : ipdot,
+                            'Type' : 'ip',
+                            'Score' : 3,
+                            'isTypedIndicator': True,
+                            'Malicious' : {
+                                'Vendor' : 'ThreatX',
+                                'Description' : 'ThreatX risk score is ' + str(e_risk),
+                                'IP' : ipdot
+                            }
+                        })
+                elif e_risk > 10:
+                    dbot_scores.append({
+                            'Vendor' : 'ThreatX',
+                            'Indicator' : ipdot,
+                            'Type' : 'ip',
+                            'Score' : 2,
+                            'isTypedIndicator': True
+                        })
+                else:
+                    dbot_scores.append({
+                            'Vendor' : 'ThreatX',
+                            'Indicator' : ipdot,
+                            'Type' : 'ip',
+                            'Score' : 0,
+                            'isTypedIndicator': True
+                        })
+
+            ec = {
```

Comment:
Please add `IP` object to the context output.

Reply:
Think i fixed this right

Reply:
The `IP` object will be added only `if dbscore == 3`, while it should be added in any case

Reply:
done